ocs_ci/ocs/ui/helpers_ui.py

```diff
@@ -95,6 +98,157 @@ def format_locator(locator, string_to_insert):
+    logger.info("Storage Class Storage Pool")
+    base_ui_obj.do_click(pvc_loc["storage-pool"])
+    base_ui_obj.do_click(pvc_loc["ceph-block-pool"])
+    if encryption:
+        logger.info("Storage Class with Encryption")
+        base_ui_obj.do_click(pvc_loc["encryption"])
+        logger.info("Checking if 'Change connection details' option is available")
+        conn_details = base_ui_obj.check_element_text(
+            expected_text="Change connection details"
+        )
+        if conn_details:
+            logger.info("Click on Change Connection Details")
+            base_ui_obj.do_click(pvc_loc["connections-details"])
+        if ocp_version >= "4.9":
+            logger.info("Click on Create new KMS connection")
+            base_ui_obj.do_click(pvc_loc["new_kms"])
+        logger.info("Storage Class Service Name")
+        base_ui_obj.do_clear(pvc_loc["service-name"])
+        base_ui_obj.do_send_keys(pvc_loc["service-name"], service_name)
+        logger.info("Storage Class Address")
+        base_ui_obj.do_clear(pvc_loc["kms-address"])
+        base_ui_obj.do_send_keys(pvc_loc["kms-address"], kms_address)
+        logger.info("Storage Class Port")
+        base_ui_obj.do_clear(pvc_loc["kms-port"])
+        base_ui_obj.do_send_keys(pvc_loc["kms-port"], "8200")
+        logger.info("Click on Advanced Settings")
+        base_ui_obj.do_click(pvc_loc["advanced-settings"])
+        logger.info("Enter Backend Path")
+        base_ui_obj.do_clear(pvc_loc["backend-path"])
+        base_ui_obj.do_send_keys(pvc_loc["backend-path"], backend_path)
+        logger.info("Enter TLS Server Name")
+        base_ui_obj.do_clear(pvc_loc["tls-server-name"])
+        base_ui_obj.do_send_keys(pvc_loc["tls-server-name"], tls_server_name)
+        logger.info("Enter Vault Enterprise Namespace")
+        base_ui_obj.do_clear(pvc_loc["vault-enterprise-namespace"])
+        base_ui_obj.do_send_keys(pvc_loc["vault-enterprise-namespace"], vault_namespace)
+        logger.info("Selecting CA Certificate")
+        ca_cert_pem = base_ui_obj.driver.find_element(
+            By.XPATH, "(//input[@type='file'])[1]"
+        )
+        ca_cert_pem.send_keys(os.path.abspath(constants.VAULT_CA_CERT_PEM))
+        logger.info("Selecting Client Certificate")
+        client_cert_pem = base_ui_obj.driver.find_element(
+            By.XPATH, "(//input[@type='file'])[2]"
+        )
+        client_cert_pem.send_keys(os.path.abspath(constants.VAULT_CLIENT_CERT_PEM))
+        logger.info("Selecting Client Private Key")
+        client_private_key_pem = base_ui_obj.driver.find_element(
+            By.XPATH, "(//input[@type='file'])[3]"
+        )
+        client_private_key_pem.send_keys(os.path.abspath(constants.VAULT_PRIVKEY_PEM))
+        base_ui_obj.take_screenshot()
+        logger.info("Saving Key Management Service Advanced Settings")
+        base_ui_obj.do_click(pvc_loc["save-advanced-settings"], enable_screenshot=True)
```

Comment:
Since these are optional values, it would be good to add conditions to use/set them